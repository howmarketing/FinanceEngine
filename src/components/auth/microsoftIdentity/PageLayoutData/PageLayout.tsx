import React from "react";
import Navbar from "react-bootstrap/Navbar";
import { useIsAuthenticated } from "@azure/msal-react";
import Center from './Center/CenterPageLayout';
import { SignInButton } from "../SignInButton/SignInButton";
import { SignOutButton } from "../SignOutButton/SignOutButton";
/**
 * Renders the navbar component with a sign-in or sign-out button depending on whether or not a user is authenticated
 * @param props
 */
export const PageLayout: React.FC<{ [x: string]: any }> = (props) => {
	const isAuthenticated = useIsAuthenticated();
	return (
		<>
			<Navbar bg="primary" variant="dark">
				<a className="navbar-brand" href="/">Microsoft Identity Platform</a>
				{isAuthenticated ? <SignOutButton /> : <SignInButton />}
			</Navbar>
			<h5>
				<Center>
					Welcome to the Microsoft Authentication Library For Javascript - React Quickstart
				</Center>
			</h5>
			<br />
			<br />
			{props.children}
		</>
	);
};
