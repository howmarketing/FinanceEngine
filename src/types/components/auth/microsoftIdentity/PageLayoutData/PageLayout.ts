export type IPageLayoutData = {
	children?: React.ReactNode;
}
