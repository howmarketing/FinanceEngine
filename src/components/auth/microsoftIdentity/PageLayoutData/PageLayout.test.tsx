import React from 'react';
import TestRenderer from 'react-test-renderer';
import { PageLayout } from "components/auth/microsoftIdentity/PageLayoutData/PageLayout";

it('Page layout component render', () => {
	const tree = TestRenderer.create(<PageLayout />).toJSON()
	expect(tree).toMatchSnapshot()
})
