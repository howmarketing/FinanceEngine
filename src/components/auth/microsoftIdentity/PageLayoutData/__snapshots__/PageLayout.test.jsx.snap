// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page layout component render 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-primary"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Microsoft Identity Platform
    </a>
    <div
      className="ml-auto dropstart"
    >
      <button
        aria-expanded={false}
        className="dropdown-toggle btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign In
      </button>
    </div>
  </nav>,
  <h5>
    <div
      className="d-flex justify-content-center"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      Welcome to the Microsoft Authentication Library For Javascript - React Quickstart
    </div>
  </h5>,
]
`;
